src/messaging/ExchangeMgr.cpp

```diff
@@ -219,15 +219,20 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
                 ec->SetMsgRcvdFromPeer(true);
             }
 
+            MessageFlags msgFlags;
+            if (isDuplicate == DuplicateMessage::kYes)
+            {
+                msgFlags.Set(MessageFlagValues::kDuplicateMessage);
+            }
             // Matched ExchangeContext; send to message handler.
-            ec->HandleMessage(packetHeader, payloadHeader, source, std::move(msgBuf));
+            ec->HandleMessage(packetHeader, payloadHeader, source, msgFlags, std::move(msgBuf));
             found = true;
             return false;
         }
         return true;
     });
 
-    if (found)
+    if (found || isDuplicate == DuplicateMessage::kYes)
```

Comment:
If no exchange found for the duplicated message, we can not simply drop the message since it may need ack.

Reply:
Updated the code.